velox/functions/lib/string/StringImpl.h

```diff
@@ -147,6 +147,36 @@ FOLLY_ALWAYS_INLINE int32_t charToCodePoint(const T& inputString) {
   return codePoint;
 }
 
+/// Returns the sequence of Unicode code point of characters in the an input
+/// string.
+template <bool isAscii, typename T>
+FOLLY_ALWAYS_INLINE std::vector<int32_t> stringToCodePoint(
+    const T& inputString) {
+  std::vector<int32_t> res;
+  auto length = inputString.size();
+
+  if constexpr (isAscii) {
+    const char* inputStringData = (char*)inputString.data();
+    for (int i = 0; i < length; i++) {
+      res.push_back((int)inputStringData[i]);
+    }
+    return res;
+  }
+
+  auto inputIndex = 0;
+  while (inputIndex < length) {
+    utf8proc_int32_t codepoint;
+    int size;
+    codepoint = utf8proc_codepoint(
+        inputString.data() + inputIndex, inputString.data() + length, size);
+    VELOX_USER_CHECK_GE(
+        codepoint, 0, "Invalid UTF-8 encoding in characters: {}", inputString);
```

Comment:
inputString may be large and contain non-printable characters. Let's specify the index at which we encountered invalid character and include maybe up to 12 bytes of the string starting from the invalid byte.